editoast/src/map/layers.rs

```diff
@@ -33,14 +33,36 @@ pub struct Layer {
     pub attribution: Option<String>,
 }
 
-#[derive(Serialize, Deserialize, PartialEq, Debug)]
+#[derive(Debug, PartialEq)]
 pub struct MapLayers {
     pub layers: HashMap<String, Layer>,
+    pub max_zoom: u8,
+}
+
+/// Parses file containing layers' description
+fn parse() -> HashMap<String, Layer> {
+    #[derive(Deserialize)]
+    struct Layers {
+        layers: HashMap<String, Layer>,
+    }
+
+    let Layers { layers } = serde_yaml::from_str(include_str!("../../map_layers.yml"))
+        .expect("static data should be valid");
```

Comment:
This doesn't work?

```suggestion
    let layers: HashMap<String, Layer> = serde_yaml::from_str(include_str!("../../map_layers.yml"))
        .expect("static data should be valid");
```

Reply:
Sadly no:

```yaml
layers:
  track_sections:
    table_name: infra_layer_track_section
    id_field: id
    views:
      geo:
        on_field: geographic
  ...
```

I can remove the indirection in the YAML though, since `layer` is the only prop at top-level.

Reply:
ok, don't bother, I didn't have the entire context and I understand now.